### PR TITLE
remove flatten.using_ints, linalg_*, linear, log_softmax.int, logdet, special_* from xfail list

### PR DESCRIPTION
**Pull request by guilhermeleobas**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #110985

cc @zou3519 @Chillee @samdow @kshitij12345 @janeyx99

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

File: aten/src/ATen/functorch/BatchRulesLinearAlgebra.cpp [hunk 1]
```diff
@@ -605,13 +602,11 @@ TORCH_LIBRARY_IMPL(aten, FuncTorchBatched, m) {
   VMAP_SUPPORT(dot, dot_batch_rule);
   VMAP_SUPPORT(mv, mv_batch_rule);
   VMAP_SUPPORT(mm, mm_batch_rule);
-  m.impl("linear", linear_decomp);
```
Comment (zou3519) [inline aten/src/ATen/functorch/BatchRulesLinearAlgebra.cpp:608]: Can we delete the linear_decomp function?

File: aten/src/ATen/functorch/BatchRulesLinearAlgebra.cpp [hunk 2]
```diff
@@ -605,13 +602,11 @@ TORCH_LIBRARY_IMPL(aten, FuncTorchBatched, m) {
...
   VMAP_SUPPORT(linalg_lu_solve, linalg_lu_solve_batch_rule);
   VMAP_SUPPORT(linalg_householder_product, householder_product_batch_rule);
   VMAP_SUPPORT(cholesky_solve, cholesky_solve_batch_rule);  // custom dim error
   VMAP_SUPPORT(linalg_lstsq, linalg_lstsq_batch_rule);  // custom errors and sometimes empty return
   VMAP_SUPPORT(linalg_lu_factor_ex, linalg_lu_factor_ex_batch_rule);
-  VMAP_SUPPORT(linalg_lu_factor, linalg_lu_factor_batch_rule);
```
Comment (zou3519) [inline aten/src/ATen/functorch/BatchRulesLinearAlgebra.cpp:614]: Can we delete linalg_lu_factor_batch_rule?

**Review by zou3519 — APPROVED** [review]
LGTM, minor questions on if we can delete more code

**Pull request closed without merging**